SWLOR.Game.Server/Feature/AIDefinition/GenericAIDefinition.cs

```diff
@@ -32,485 +34,220 @@ static float CalculateAverageHP(uint creature)
-            {
-                return (FeatType.DoubleThrust3, target);
-            }
-            if (CheckIfCanUseFeat(self, target, FeatType.DoubleThrust2))
-            {
-                return (FeatType.DoubleThrust2, target);
-            }
-            if (CheckIfCanUseFeat(self, target, FeatType.DoubleThrust1))
-            {
-                return (FeatType.DoubleThrust1, target);
-            }
-
-
-            // Leg Sweep
-            if (CheckIfCanUseFeat(self, self, FeatType.LegSweep3))
-            {
-                return (FeatType.LegSweep3, self);
-            }
-            if (CheckIfCanUseFeat(self, self, FeatType.LegSweep2))
-            {
-                return (FeatType.LegSweep2, self);
-            }
-            if (CheckIfCanUseFeat(self, self, FeatType.LegSweep1))
-            {
-                return (FeatType.LegSweep1, self);
-            }
-
-            // Cross Cut
-            if (CheckIfCanUseFeat(self, target, FeatType.CrossCut3))
-            {
-                return (FeatType.CrossCut3, target);
-            }
-            if (CheckIfCanUseFeat(self, target, FeatType.CrossCut2))
-            {
-                return (FeatType.CrossCut2, target);
-            }
-            if (CheckIfCanUseFeat(self, target, FeatType.CrossCut3))
-            {
-                return (FeatType.CrossCut1, target);
-            }
-
-            // Circle Slash
-            if (CheckIfCanUseFeat(self, self, FeatType.CircleSlash3))
-            {
-                return (FeatType.CircleSlash3, self);
-            }
-            if (CheckIfCanUseFeat(self, self, FeatType.CircleSlash2))
-            {
-                return (FeatType.CircleSlash2, self);
-            }
-            if (CheckIfCanUseFeat(self, self, FeatType.CircleSlash1))
-            {
-                return (FeatType.CircleSlash1, self);
-            }
-
-            // Double Strike
-            if (CheckIfCanUseFeat(self, target, FeatType.DoubleStrike3))
-            {
-                return (FeatType.DoubleStrike3, target);
-            }
-            if (CheckIfCanUseFeat(self, target, FeatType.DoubleStrike2))
-            {
-                return (FeatType.DoubleStrike2, target);
-            }
-            if (CheckIfCanUseFeat(self, target, FeatType.DoubleStrike1))
-            {
-                return (FeatType.DoubleStrike1, target);
-            }
 
-            // Electric Fist
-            if (CheckIfCanUseFeat(self, self, FeatType.ElectricFist3))
-            {
-                return (FeatType.ElectricFist3, self);
-            }
-            if (CheckIfCanUseFeat(self, self, FeatType.ElectricFist2))
-            {
-                return (FeatType.ElectricFist2, self);
-            }
-            if (CheckIfCanUseFeat(self, self, FeatType.ElectricFist1))
-            {
-                return (FeatType.ElectricFist1, self);
+            List<FeatType> featTypes = Enum.GetValues(typeof(FeatType))
```

Comment:
There's a hierarchy to the AI decisions being made here which aren't preserved by making a list out of it. I.E the AI is intended to heal itself prior to doing attacks, for example. I'm just getting into this PR but I wanted to mention it as I think this changes their behavior. 

Additionally, this AI process occurs very often - I believe on heartbeat - so the iteration over every single feat type is going to be too expensive. If this type of approach is to be used, we need to whitelist the ability types to look for rather than checking every single one of them on the NPC.